app/helpers/annotations_helper.rb

```
@@ -0,0 +1,99 @@
+      file_annotations = []
+      annotations.each do |description, value, line, user, id, position, filename, global|
+        annotation = Annotation.find_by(id: id)
+        if annotation && annotation.problem_id == problem_id && annotation.rubric_item_id.nil?
+          file_annotations << [description, value, line, user, id, position, filename, global]
+        end
+      end
+      filtered_annotations_by_file[filename] = file_annotations unless file_annotations.empty?
+    end
+    
+    [filtered_global_annotations, filtered_annotations_by_file]
+  end
```

RuboCop (1.73)

[convention] Assignment Branch Condition size for get_non_rubric_annotations is too high. [<25, 15, 12> 31.53/23]

(Metrics/AbcSize)